source/extensions/tracers/zipkin/zipkin_core_types.cc

```diff
@@ -219,6 +238,41 @@ const std::string Span::toJson() {
   return json_string;
 }
 
+const zipkin::proto3::Span Span::toProtoSpan() const {
+  zipkin::proto3::Span span;
+  span.set_trace_id(traceIdAsByteString());
+  span.set_parent_id(parentIdAsByteString());
+  span.set_id(idAsByteString());
+  span.set_name(name_);
+
+  if (timestamp_) {
```

Comment:
curious how this can be unset. if this is a side effect of shared server span handling, we should follow-up on that.

For example, in v1 we did some trick... if you accepted a span ID from X-B3-SpanId header, we didn't set timestamp or duration. If this is null on account of that, we should set shared =true and then also backfill timestamp and duration as in v2 we always set timestamp and duration when we have it.

Reply:
I'm not sure, this guard was in-place for `toJson()` https://github.com/envoyproxy/envoy/blob/fa0b52a874b67c857bd314620592d39280767330/source/extensions/tracers/zipkin/zipkin_core_types.cc#L190. Will investigate more on that.